syntax = 'proto3';

package strike;

import 'google/protobuf/timestamp.proto';


service ConflictsService {
  rpc CreateConflict(Conflict) returns (SingleConflictResponse) {}
}

enum ConflictTypes {
  UNSPECIFIED = 0;
  PAYMENT = 1;
  LAYOFF = 2;
  LIQUIDATION = 3;
  OTHER = 4;
  DISMISSAL = 5;
  MANAGEMENT_POLITICS = 6;
  WORK_CONDITIONS = 7;
  LABOR_HOURS = 8;
  COLLECTIVE_AGREEMENT = 9;
  PAYMENT_DELAY = 10;
  LABOUR_RIGHTS = 11;
}


message Conflict {
  optional int64 id = 1;
  ConflictTypes type = 2;
  Duration duration = 3;
  int64 enterprise_id = 4;
  string description = 5;
  optional string results = 6;
  optional float success_rate = 7;
}

message SingleConflictResponse {
  Conflict conflict = 1;
  ConflictExtraData extra_data = 2;
}

message ConflictExtraData {
  optional google.protobuf.Timestamp updated_at = 1;
  google.protobuf.Timestamp created_at = 2;
}

message Duration {
  google.protobuf.Timestamp lower = 1;
  optional google.protobuf.Timestamp upper = 2;
}

